### PR TITLE
User can continue adding cards

### PR DESCRIPTION
**Pull request by StevenNagata**
react-flashcards-issue4

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -7,10 +7,12 @@ import FlashcardForm from './flashcard-form'
 export default class App extends React.Component {
   constructor(props) {
     super(props)
+    const stateJson = localStorage.getItem('current-app-state')
+    const appState = JSON.parse(stateJson) || {}
     const path = hash.parse(location.hash).path
     this.state = {
-      view: { path },
-      flashcards: []
+      view: appState.path || { path },
```
Comment (thebearingedge) [inline src/app.js]: Let your user's browser dictate the view according to the `location.hash`.

File: src/app.js [hunk 2]
```diff
@@ -36,6 +38,11 @@ export default class App extends React.Component {
         view: { path }
       })
     })
+    window.addEventListener('beforeunload', () => {
+      const { view, flashcards } = this.state
```
Comment (thebearingedge) [inline src/app.js]: The browser's history (back/forward) will keep track of the view, so you don't need to persist it.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Comment by StevenNagata** [comment]
Updated gif looks the same

**Review by thebearingedge — APPROVED** [review]
Yep.

**Pull request merged**